### PR TITLE
Add a new `generate-github-app-token-from-vault-secrets` composite GHA

### PR DESCRIPTION
**Pull request by clementnero**
Related to https://github.com/camunda/team-infrastructure/issues/204

Add a new composite GitHub action to generate a GitHub token from Github App secrets (ID & Private Key) stored in Vault. It is part of the refactoring of the `maintenance_project.yml` workflow.

### CODE REVIEW

File: generate-github-app-token-from-vault-secrets/action.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  token:
+    description: The generated GitHub token
+    value: ${{ steps.github-token.outputs.token }}
+
+runs:
+  using: composite
+  steps:
+  - name: Import the GitHub App ID from vault
+    id: app-id
+    uses: hashicorp/vault-action@v2.4.3
```
Comment (maxdanilov) [inline generate-github-app-token-from-vault-secrets/action.yml:49]:
```suggestion
    uses: hashicorp/vault-action@v2.5.0
```

let's use the recent version of it

File: generate-github-app-token-from-vault-secrets/action.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          format(
+            '{0} {1} | {2};',
+            inputs.github-app-id-vault-path,
+            inputs.github-app-id-vault-key,
+            'id'
+          ) 
+        }}
+  - name: Import the GitHub App secret key from vault
+    id: app-private-key
+    uses: hashicorp/vault-action@v2.4.3
```
Comment (maxdanilov) [inline generate-github-app-token-from-vault-secrets/action.yml:67]:
```suggestion
    uses: hashicorp/vault-action@v2.5.0
```

let's use the recent version of it

**Review by maxdanilov — APPROVED** [review]
Looks good to me, a really nice and clean implementation.
Just one suggestion to use the current version of the `vault-action`

File: generate-github-app-token-from-vault-secrets/action.yml (hunk 1 above)
Comment (clementnero) [reply generate-github-app-token-from-vault-secrets/action.yml:49]: Good point!

**Pull request merged**